src/main/java/dukes/engine/Duke.java

```diff
@@ -0,0 +1,106 @@
+package dukes.engine;
+
+import dukes.util.DukeException;
+import dukes.util.TaskList;
+import dukes.util.UI;
+import dukes.util.Storage;
+import dukes.util.Parser;
+
+import dukes.command.Command;
+
+import java.util.Scanner;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * The main driver class of Duke software.
+ */
+//public class Duke extends Application {
+public class Duke {
+
+    /** Handles loading and saving information to hard disk. */
+    private Storage storage;
+    /** Contains task list. */
+    private TaskList tasks;
+    /** Interprets the user command. */
+    private Parser parser;
+    /** Handles interactions with user. */
+    private UI ui;
+    public static final String DEFAULT_PATH = "data/tasks.txt";
+
+    // When JavaFX builds the Application, "Application.launch(Duke.class, args)",
+    // it is actually trying to create Duke().
+    // But if your code does not have this constructor, it will fail.
+    public Duke() {
+        this(DEFAULT_PATH);
+    }
+
+    /**
+     * Constructor of Duke class.
+     * Handles the loading error when Duke starts.
+     *
+     * @param filePath the file path of the data file.
+     */
+    public Duke(String filePath) {
+        ui = new UI();
+        parser = new Parser();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError(e.getMessage());
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Runner of Duke class. Call classes to interpret command and execute.
+     * Catch all the runtime DukeExceptions and provide feedback.
+     */
+    public void run() {
```

Comment:
I think since we are using the GUI and FXML for the handling of inputs, there is no longer a need for the use of Scanners since the front end of the GUI will call the handleUserInput() method in MainWindow.java. Perhaps this could be removed?